packages/combobox/src/Combobox.tsx

```diff
@@ -1059,19 +1062,37 @@ export default function Combobox<M extends boolean>({
    * Global Event Handler
    *
    */
-  // Global backdrop click handler
-  const handleBackdropClick = ({ target }: MouseEvent) => {
-    const isChildFocused =
-      menuRef.current?.contains(target as Node) ||
-      comboboxRef.current?.contains(target as Node) ||
-      false;
 
-    if (!isChildFocused) {
-      setOpen(false);
+  /**
+   * We add two event handlers to the document to handle the backdrop click behavior.
+   * Intended behavior is to close the menu, and keep focus on the Combobox.
+   * No other click event handlers should fire on backdrop click
+   *
+   * 1. Mousedown event fires
+   * 2. We prevent `mousedown`'s default behavior, to prevent focus from being applied to the body (or other target)
+   * 3. Click event fires
+   * 4. We handle this event on _capture_, and stop propagation before the `click` event propagates all the way to any other element.
+   *  This ensures that even if we click on a button, that handler is not fired
+   * 5. Then we call `closeMenu`, setting `isOpen = false`, and rerender the component
+   */
+  useEventListener('mousedown', (mousedown: MouseEvent) => {
```

Comment:
Do we see it problematic that we're adding a new event listener for every Combobox component on the page? Usually we're protected by React using event delegation under the hood, but this doesn't use React synthetic events.

Reply:
Yeah, that's a fair concern. I think the solution to that would be to do our own event delegation within `useEventHandler`, no?

Reply:
Unless there's a way to add a React Synthetic event to the document (or react root) programmatically like this? I don't know of a way to do that though

Reply:
Talked offline – this event listener should only be registered when the combobox is open.